src/wifi.cpp

```diff
@@ -0,0 +1,694 @@
+        // which default values are show_hidden:false, scan_type:active, scan_time.active.min:0,
+        // scan_time.active.max:120 milliseconds, scan_time.passive:360 milliseconds
+        // home_chan_dwell_time:30ms
+        /* const wifi_scan_config_t *config = */ NULL,
+        // if block is true, this API will block the caller until the scan is done, otherwise it will return immediately
+        /* bool block = */ true));
+
+    // Stop if it takes longer than 10 seconds
+    vTaskDelay(NULL, 10,000 ms);
+    ESP_ERROR_CHECK(esp_wifi_scan_stop());
+
+    // Get the number of AP found
+    uint16_t num_ap_found = 0;
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_num(/* uint16_t *number = */ num_ap_found));
+    wifi_ap_record_t *ap_record = new wifi_ap_record[num_ap_found];
+
+    // Get record of all APs found (getting removes APs)
+    ESP_ERROR_CHECK(esp_wifi_scan_get_ap_records(
+	/* uint16_t *number = */ num_ap_found,
+        /* wifi_ap_record_t *ap_records = */ ap_records));
+    delete [] ap_record;
+
+    // Free memory storing last AP search results
+    ESP_ERROR_CHECK(esp_wifi_clear_ap_list());
+
+    // Get AP currently connected to (NOTE: shouldn't abort on fail, WHO is it connecting to???) 
+    ESP_ERROR_CHECK(esp_wifi_sta_get_ap_info(/* wifi_ap_record_t *ap_info = */ &ap_info));
+
+    // Set MAC address of this station (this can only be caled when the intterface is disabled?)
+    // ESP_ERROR_CHECK(esp_wifi_set_mac(
+    //    /* wifi_interface_t ifx = */,
+    //    /* const uint8_t mac[6] = */);
+
+    // Get MAC address of this station
+    ESP_ERROR_CHECK(esp_wifi_get_mac(
+        /* wifi_interface_t ifx = */ ,
+        /* uint8_t mac[6] = */ mac_addr);
+}
+#endif
+
+// Allocate resources needed for TCP/IP/WiFi
+static inline bool wifi_init()
+{
+    // Save status from checks
+    // If any check fails, do none of the following
+    esp_err_t status = ESP_OK;
+
+    // Initialize the underlying TCP/IP stack
+    ESP_ERROR_CHECK_WITHOUT_ABORT(status = esp_netif_init());
+    if (ESP_OK != status) return false;
+
+    // Initialize the default ESP event loop
+    ESP_ERROR_CHECK_WITHOUT_ABORT(status = esp_event_loop_create_default());
+    if (ESP_OK != status) return false;
+
+    // Create default WiFi station (something that connects to a router,
+    // instead of serving as a router itself) in the WiFi driver
+    esp_netif = esp_netif_create_default_wifi_sta();
+    ESP_ERROR_CHECK_WITHOUT_ABORT(status = (nullptr != esp_netif) ? ESP_OK : ESP_FAIL);
+    if (ESP_OK != status) return false;
+
+    // Initialize and allocate WiFi structures and start its task with default settings,
+    // We can worry about getting the correct credentials and authentication method later
+    const wifi_init_config_t wifi_init_config = WIFI_INIT_CONFIG_DEFAULT();
+    ESP_ERROR_CHECK_WITHOUT_ABORT(status = esp_wifi_init(/* const wifi_init_config_t *config = */ &wifi_init_config));
+    if (ESP_OK != status) return status;
```

Comment:
return `false` here not `status`

Reply:
addressed in: https://github.com/jCallon/squirt/pull/1/commits/e37aa7221d9c98ca47cf9bfe7d147cfb70de036c